<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QEllipseSlider</class>
 <widget class="QWidget" name="QEllipseSlider">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>431</width>
    <height>216</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="mainGroup">
     <property name="title">
      <string>GroupBox</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="valueLayout">
        <item>
         <widget class="QSlider" name="valueSlider">
          <property name="minimum">
           <number>400</number>
          </property>
          <property name="maximum">
           <number>3000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="valueSpinBox">
          <property name="suffix">
           <string>cm</string>
          </property>
          <property name="minimum">
           <number>400</number>
          </property>
          <property name="maximum">
           <number>3000</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="lockValueButton">
          <property name="text">
           <string>Lock</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="motorGroup">
        <property name="title">
         <string>Motors</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="sideMotorLabel">
           <property name="text">
            <string>Side</string>
           </property>
           <property name="buddy">
            <cstring>sideMotorSlider</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSlider" name="sideMotorSlider">
           <property name="maximum">
            <number>80</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QSpinBox" name="sideMotorSpinBox">
           <property name="suffix">
            <string>mm</string>
           </property>
           <property name="maximum">
            <number>300</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="frontMotorLabel">
           <property name="text">
            <string>Front</string>
           </property>
           <property name="buddy">
            <cstring>frontMotorSlider</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSlider" name="frontMotorSlider">
           <property name="maximum">
            <number>80</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QSpinBox" name="frontMotorSpinBox">
           <property name="suffix">
            <string>mm</string>
           </property>
           <property name="maximum">
            <number>300</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="warningLabel">
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; color:#ff0000;&quot;&gt;Unreachable perimeter&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>valueSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>valueSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>52</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valueSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>valueSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>54</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideMotorSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>QEllipseSlider</receiver>
   <slot>setSideMotorValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>162</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontMotorSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>QEllipseSlider</receiver>
   <slot>setFrontMotorValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>valueSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>QEllipseSlider</receiver>
   <slot>onPerimeterChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>49</y>
    </hint>
    <hint type="destinationlabel">
     <x>318</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideMotorSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sideMotorSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>116</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideMotorSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sideMotorSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>375</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontMotorSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>frontMotorSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>375</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontMotorSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>frontMotorSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>376</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sideMotorSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEllipseSlider</receiver>
   <slot>setSideMotorValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>373</x>
     <y>109</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>32</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>frontMotorSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QEllipseSlider</receiver>
   <slot>setFrontMotorValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>361</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>427</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setSideMotorValue(int)</slot>
  <slot>setFrontMotorValue(int)</slot>
  <slot>onPerimeterChanged(int)</slot>
 </slots>
</ui>
